src/gboml/scope.py

```diff
@@ -33,9 +34,27 @@ def _add_to_scope(self, ast, wrapper=lambda x: x, whenPresent: OverrideBehavior
             else:
                 pass
 
+        if ast.name in ['T', 't']:
+            raise KeyError(f"Identifier {ast.name} cannot be redifined (reserved keyword)")
+
         self.content[ast.name] = wrapper(create_scope(ast, self))
         return self.content[ast.name]
 
+    def _add_unnamed_to_scope(self, name, whenPresent: OverrideBehavior = OverrideBehavior.fail) -> None:
+        if name in self.content:
+            print(self.content)
+            if whenPresent == OverrideBehavior.fail:
+                raise RuntimeError(f"Identifier {name} is already used")
+            elif whenPresent == OverrideBehavior.ignore:
+                return
+            else:
+                pass
+
+        if name in ['T', 't']:
+            raise KeyError(f"Identifier {ast.name} cannot be redifined (reserved keyword)")
+
+        self.content[name] = {}
```

Comment:
The empty dict is used so when we try to get a name in a scope (`scope['MyNode']`), we get a KeyError if MyNode doesn't exist (while using a None as value instead would give another error). Is this OK?